api/lib/infrastructure/repositories/campaign-participation-overview-repository.js

```diff
@@ -0,0 +1,117 @@
+const { knex } = require('../bookshelf');
+const Assessment = require('../../domain/models/Assessment');
+const Campaign = require('../../domain/models/Campaign');
+const CampaignParticipationOverview = require('../../domain/read-models/CampaignParticipationOverview');
+const { fetchPage } = require('../utils/knex-utils');
+
+module.exports = {
+
+  async findAllByUserId(userId, page) {
+    const queryBuilder = _findByUserId({ userId });
+    const { results, pagination } = await _paginateQuery(queryBuilder, page);
+    const campaignParticipationOverviews = _toReadModel(results);
+
+    return {
+      campaignParticipationOverviews,
+      pagination,
+    };
+  },
+
+  async findByUserIdWithFilters({ userId, states, page }) {
+    const queryBuilder = _findByUserId({ userId });
+
+    if (states.includes('ENDED')) {
+      queryBuilder.modify(_filterWhithEndedState, states);
+    } else {
+      queryBuilder.modify(_filterByAssessmentStates, buildAssessementStates(states));
+      queryBuilder.modify(_filterBySharedState, false);
+      queryBuilder.whereNull('campaigns.archivedAt');
+    }
+
+    const { results, pagination } = await _paginateQuery(queryBuilder, page);
+    const campaignParticipationOverviews = _toReadModel(results);
+
+    return {
+      campaignParticipationOverviews,
+      pagination,
+    };
+  },
```

Comment:
Je propose de n'avoir qu'une seule méthode : 

```suggestion
  async findByUserIdWithFilters({ userId, states, page }) {
    const queryBuilder = _findByUserId({ userId });

    if (states && states.length > 0) {
      if (states.includes('ENDED')) {
        queryBuilder.modify(_filterWhithEndedState, states);
      } else {
        queryBuilder.modify(_filterByAssessmentStates, buildAssessementStates(states));
        queryBuilder.modify(_filterBySharedState, false);
        queryBuilder.whereNull('campaigns.archivedAt');
      }
    }

    const { results, pagination } = await _paginateQuery(queryBuilder, page);
    const campaignParticipationOverviews = _toReadModel(results);

    return {
      campaignParticipationOverviews,
      pagination,
    };
  },
```

Ce qui permettrait d'avoir dans le use-case : 

```
module.exports = async function findUserCampaignParticipationOverviews({ userId, states, campaignParticipationOverviewRepository, page }) {
  const newStates = [].concat(states)
  return campaignParticipationOverviewRepository.findByUserIdWithFilters({ userId, states: newStates, page });
};

```